retrofit/src/main/java/retrofit/Converter.java

```diff
@@ -40,10 +51,23 @@
 
     /**
      * Returns a {@link Converter} for converting {@code type} to an HTTP request body, or null if
-     * {@code type} cannot be handled by this factory.
+     * {@code type} cannot be handled by this factory. This is used to create converters for types
+     * specified by {@link Body @Body}, {@link Part @Part}, and {@link PartMap @PartMap}
+     * values.
      */
     public Converter<?, RequestBody> toRequestBody(Type type, Annotation[] annotations) {
       return null;
     }
+
+    /**
+     * Returns a {@link Converter} for converting {@code type} to a {@link String}, or null if
+     * {@code type} cannot be handled by this factory. This is used to create converters for types
+     * specified by {@link Field @Field}, {@link FieldMap @FieldMap} values,
+     * {@link Header @Header}, {@link Path @Path}, {@link Query @Query}, and
+     * {@link QueryMap @QueryMap} values.
+     */
+    public Converter<?, String> toString(Type type, Annotation[] annotations) {
```

Comment:
Yeah, pretty badly. Anything in `Object` is going to cause problems with nested classes, etc.